pkg/api/vlabs/validate.go

```diff
@@ -505,32 +505,34 @@ func (a *KubernetesConfig) Validate(k8sVersion string) error {
 			if err != nil {
 				return fmt.Errorf("--node-status-update-frequency '%s' is not a valid duration", val)
 			}
-			if a.CtrlMgrNodeMonitorGracePeriod == "" {
-				return fmt.Errorf("--node-status-update-frequency was set to '%s' but OrchestratorProfile.KubernetesConfig.CtrlMgrNodeMonitorGracePeriod was not set", val)
+			if a.ControllerManagerConfig != nil {
+				if a.ControllerManagerConfig["--node-monitor-grace-period"] == "" {
```

Comment:
Do we still need this check and some other checks below since we provide a default value?

Reply:
Yep, you're right, addressing in a follow-up commit.